### PR TITLE
Format unnamed function arguments

### PR DESCRIPTION
**Pull request by marcusklaas**
This closes https://github.com/nrc/rustfmt/issues/145.

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -780,10 +786,23 @@ impl<'a> FmtVisitor<'a> {
...
+                    pprust::ty_to_string(&arg.ty))
+        } else {
+            pprust::ty_to_string(&arg.ty)
+        }
+    }
+}
+
+fn is_named(arg: &ast::Arg) -> bool {
+    if let ast::Pat_::PatIdent(_, ident, _) = arg.pat.node {
+        if ident.node == token::special_idents::invalid {
```
Comment (nrc) [inline src/items.rs]: Can replace the whole if stmt with `ident.node != token::special_idents::invalid`

Comment (nrc) [inline src/items.rs]: Would then need an `else` further down

File: src/items.rs [hunk 2]
```diff
@@ -292,7 +292,13 @@ impl<'a> FmtVisitor<'a> {
                                      args[min_args-1..].iter(),
                                      ",",
                                      ")",
-                                     |arg| arg.pat.span.lo,
+                                     |arg| {
+                                         if is_named(arg) {
+                                             arg.pat.span.lo
+                                         } else {
+                                             arg.ty.span.lo
+                                         }
```
Comment (nrc) [inline src/items.rs]: I'd pull this out into a helper function

**Comment by marcusklaas** [comment]
Amended!

**Pull request merged**